agent/routine-leak-checker/leak_test.go

```diff
@@ -15,31 +15,20 @@ import (
 )
 
 func testTLSCertificates(serverName string) (cert string, key string, cacert string, err error) {
-	ca, _, err := tlsutil.GenerateCA(tlsutil.CAOpts{})
-	if err != nil {
-		return "", "", "", err
-	}
-
-	// generate leaf
-	serial, err := tlsutil.GenerateSerialNumber()
+	signer, _, err := tlsutil.GeneratePrivateKey()
```

Comment:
I think I forgot to fix this one in `CAOpts`: the same signer needs to be used for CA and certs.